src/app/websites/website-services/afterdark/afterdark.service.ts

```diff
@@ -0,0 +1,199 @@
+import { getTags, supportsFileType } from '../../helpers/website-validator.helper';
+import * as dotProp from 'dot-prop';
+
+const ACCEPTED_FILES = ['png', 'jpeg', 'jpg'];
+
+function submissionValidate(submission: Submission, formData: SubmissionFormData): any[] {
+  const problems: any[] = [];
+  if (!supportsFileType(submission.fileInfo, ACCEPTED_FILES)) {
+    problems.push(['Does not support file format', { website: 'Twitter', value: submission.fileInfo.type }]);
+  }
+  const tags = getTags(submission, AfterDark.name);
+  if (tags.length < 2) {
+    problems.push(['Requires minimum tags', { website: 'AfterDark', value: 2 }]);
+  }
+  const collections = dotProp.get(formData, 'AfterDark.options.folders', []);
+  if (collections.length < 1) {
+    problems.push(['AfterDark requires at least one collection folder']);
+  }
+  return problems;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+@Website({
+  acceptedFiles: ACCEPTED_FILES,
+  additionalFiles: false,
+  displayedName: 'AfterDark',
+  login: {
+    url: 'https://afterdark.art/login',
+  },
+  components: {
+    submissionForm: AfterDarkSubmissionForm,
+  },
+  validators: {
+    submission: submissionValidate
+  },
+  preparsers: {},
+  parsers: {
+    description: [(html: string) => html],
+  },
+})
+export class AfterDark extends BaseWebsiteService implements WebsiteService {
+  readonly BASE_URL: string = 'https://afterdark.art';
+
+  constructor(private _profileManager: LoginProfileManagerService) {
+    super();
+  }
+
+  private async getCsrfToken(profileId: string): Promise<string> {
+    let cookies = await getCookies(profileId, this.BASE_URL);
+    const response = await got.get(this.BASE_URL, this.BASE_URL, cookies, profileId);
+    if (response.statusCode !== 200) {
+      throw new Error(`Bad response getting csrf token ${response.statusCode}`);
+    }
+    const match = response.body.match(/name="csrfmiddlewaretoken" value="([^"]+)"/);
+    if (!match) {
+      throw new Error('Could not find csrf token in body');
+    }
+    const csrfToken = match[1];
+    return csrfToken;
+  }
+
+  private async getUserData(profileId: string, csrfToken: string): Promise<any> {
+    const body = JSON.stringify({
+      "operationName": "UserQuery",
+      "variables": {},
+      "query": `query UserQuery {
+        userViewer {
+          user {
+            id
+            username
+            isUserAdmin
+            isProUser
+            isEmailVerified
+            collectionSet {
+              edges {
+                node {
+                  id
+                  title
+                }
+              }
+            }
+          }
+        }
+      }`
+    });
+    let cookies = await getCookies(profileId, this.BASE_URL);
+    const response = await got.gotPost(`${this.BASE_URL}/graphql/`, {}, this.BASE_URL, cookies, {
+      body,
+      headers: {
+        'content-type': 'application/json',
+        'X-CSRFToken': csrfToken,
+      },
+    });
```

Comment:
Good to see you were able to get this to work.
Just want to verify that you tested it and it was working as expected.

Reply:
Yep!  This call is proof that the user is still logged in.

![login](https://user-images.githubusercontent.com/1375517/92552226-00361200-f215-11ea-80ee-abb8a2c14709.png)